### PR TITLE
Update creating project API to support proxy cache project

### PR DESCRIPTION
**Pull request by ywk253100**
Update creating project API to support proxy cache project

Signed-off-by: Wenkai Yin <yinw@vmware.com>

### CODE REVIEW

**Comment by reasonerjt** [comment]
Please make sure the proxy registry endpoint of a project cannot be modified.

**Comment by ywk253100** [comment]
> Please make sure the proxy registry endpoint of a project cannot be modified.

@reasonerjt It cannot be modified in the current implementation

File: src/common/models/project.go [hunk 1]
```diff
@@ -136,9 +137,10 @@ func isTrue(value string) bool {
 // List projects which user1 is member of: query := &QueryParam{Member:&Member{Name:"user1"}}
 // List projects which user1 is the project admin : query := &QueryParam{Member:&Member{Name:"user1",Role:1}}
 type ProjectQueryParam struct {
-	Name       string       // the name of project
-	Owner      string       // the username of project owner
-	Public     *bool        // the project is public or not, can be ture, false and nil
+	Name       string // the name of project
+	Owner      string // the username of project owner
+	Public     *bool  // the project is public or not, can be ture, false and nil
+	RegistryID int64
```
Comment (stonezdj) [inline src/common/models/project.go:143]: Duplicate

**Comment by reasonerjt** [comment]
I don't know if it's a good idea to fail more explicitly.  But can we add a test case and document it in the swagger?

**Pull request merged**